lib/streamlit/runtime/app_session.py

```diff
@@ -517,6 +526,9 @@ def _handle_scriptrunner_event_on_event_loop(
         page_script_hash : str | None
             A hash of the script path corresponding to the page currently being
             run. Set only for the SCRIPT_STARTED event.
+
+        fragment_id : str | None
+            The fragment ID if this script run corresponds to a fragment.
```

Comment:
nit: what do you think about rewording this slightly to `...if this script run was triggered by a fragment` or similar? My thinking here is that if the whole app re-runs, fragments are also executed in the normal flow (so it kind of corresponds to the fragment code as well), but the `fragment_id` would not be set as the run is for the whole app.

Reply:
I'm a bit concerned in general to term and explain the internal aspects too much based on the decorator usage, because I think there are a lot of important future uses of this logic that are independent of the usage of the decorator. See my comment [here](https://www.notion.so/snowflake-corp/Product-Spec-eeb691a8642d41e8969f1c82f8ac2370?pvs=4#8373e4c33789420380422ace86003632).

Reply:
Changed to @raethlein's suggested wording for now, but I agree that we'll want to consider renaming the internals of the infrastructure required to implement the decorator even if we're keeping the `st.experimental_fragment` name.

I'll defer the rename to a later time (most likely this upcoming sprint) to be able to focus on getting things working e2e first + I'm not entirely sure what the names for the internals should be at the moment (going back to `Partial*` seems reasonable, but I'm not 100% sure on that yet).